crates/oxc_mangler/src/lib.rs

```diff
@@ -129,51 +137,73 @@ impl Mangler {
         // A slot is the occurrence index of a binding identifier inside a scope.
         let mut symbol_table = symbol_table;
 
-        // Total number of slots for all scopes
-        let mut total_number_of_slots: Slot = 0;
-
         // All symbols with their assigned slots. Keyed by symbol id.
         let mut slots: Vec<'_, Slot> = Vec::with_capacity_in(symbol_table.len(), &allocator);
         for _ in 0..symbol_table.len() {
             slots.push(0);
         }
 
-        // Keep track of the maximum slot number for each scope
-        let mut max_slot_for_scope = Vec::with_capacity_in(scope_tree.len(), &allocator);
-        for _ in 0..scope_tree.len() {
-            max_slot_for_scope.push(0);
-        }
+        let mut slot_liveness: std::vec::Vec<FixedBitSet> = vec![];
 
         // Walk the scope tree and compute the slot number for each scope
         let mut tmp_bindings = std::vec::Vec::with_capacity(100);
-        for scope_id in scope_tree.descendants_from_root() {
+        for scope_id in iter::once(scope_tree.root_scope_id())
+            .chain(scope_tree.iter_all_child_ids(scope_tree.root_scope_id()))
+        {
+            let nearest_var_scope_id = scope_tree
+                .ancestors(scope_id)
+                .find(|s_id| scope_tree.get_flags(*s_id).is_var())
+                .unwrap_or(scope_tree.root_scope_id());
```

Comment:
I had this code and "treat all symbols as var for now" comment before. But it turns out this was already working so I removed these.